qiskit_quantuminspire/qi_jobs.py

```diff
@@ -46,22 +74,124 @@ def __init__(
+            await asyncio.gather(*run_coroutines)
+            await self._enqueue_batch_job(api_client, batch_job.id)
+            self.batch_job_id = batch_job.id
+
+    async def _create_project(self, api_client: ApiClient, owner_id: int) -> Project:
+        api_instance = ProjectsApi(api_client)
+        obj = ProjectIn(
+            owner_id=owner_id,
+            name=self.program_name,
+            description="Project created by SDK",
+            starred=False,
+        )
+        return await api_instance.create_project_projects_post(obj)
+
+    async def _create_algorithm(self, api_client: ApiClient, project_id: int) -> Algorithm:
+        api_instance = AlgorithmsApi(api_client)
+        obj = AlgorithmIn(
+            project_id=project_id, type=AlgorithmType.QUANTUM, shared=ShareType.PRIVATE, name=self.program_name
+        )
+        return await api_instance.create_algorithm_algorithms_post(obj)
+
+    async def _create_commit(self, api_client: ApiClient, algorithm_id: int) -> Commit:
+        api_instance = CommitsApi(api_client)
+        obj = CommitIn(
+            description="Commit created by SDK",
+            algorithm_id=algorithm_id,
+        )
+        return await api_instance.create_commit_commits_post(obj)
+
+    async def _create_file(
+        self, api_client: ApiClient, commit_id: int, language_id: int, circuit: QuantumCircuit
+    ) -> File:
+        api_instance = FilesApi(api_client)
+        obj = FileIn(
+            commit_id=commit_id,
+            content=cqasm_dumps(circuit),
+            language_id=language_id,
+            compile_stage=CompileStage.NONE,
+            compile_properties={},
+        )
+        return await api_instance.create_file_files_post(obj)
+
+    async def _create_batch_job(self, api_client: ApiClient, backend_type_id: int) -> BatchJob:
+        api_instance = BatchJobsApi(api_client)
+        obj = BatchJobIn(backend_type_id=backend_type_id)
+        return await api_instance.create_batch_job_batch_jobs_post(obj)
+
+    async def _create_job(
+        self, api_client: ApiClient, file_id: int, batch_job_id: int, number_of_shots: Optional[int] = None
+    ) -> Job:
+        api_instance = JobsApi(api_client)
+        obj = JobIn(file_id=file_id, batch_job_id=batch_job_id, number_of_shots=number_of_shots)
+        return await api_instance.create_job_jobs_post(obj)
+
+    async def _enqueue_batch_job(self, api_client: ApiClient, batch_job_id: int) -> BatchJob:
+        api_instance = BatchJobsApi(api_client)
+        return await api_instance.enqueue_batch_job_batch_jobs_id_enqueue_patch(batch_job_id)
+
+    async def _get_language(self, api_client: ApiClient, language_name: str) -> Union[Language, None]:
+        language_api_instance = LanguagesApi(api_client)
+        languages_page = await language_api_instance.read_languages_languages_get()
+        for lan in languages_page.items:
+            if language_name.lower() in lan.name.lower():
+                return lan
```

Comment:
Shouldn't we also be checking language version? Speaking of, why is cQASM listed as version 1.0 in the staging database? In case the cQASM version is incorrectly listed, I will accept this function as is, and we will make a separate ticket for fixing that.

Reply:
I agree, a proper logic for this should be a ticket by itself.